refactor(core): minor improvements

JeanMeche opened the pull request:
minor optimisation, removing unused paramters and improve tree shaking

JeanMeche committed:
refactor(core): minor improvements

minor optimisation, removing unused paramters and improve tree shaking

AndrewKushnir reviewed (approved):
Thanks for the improvements 👍

JeanMeche committed:
refactor(core): remove duplicate stringify

`throwProviderNotFoundError` already stringifies for error. Also let's restrain the paramter type.

atscott commented:
This PR was merged into the repository by commit 4cb4e984fb4bfc09cfa9ffe0e97e55116801a04d.
